### PR TITLE
refactor(connector): [trustpay] refactor trustpay and handled variants errors

### PR DESCRIPTION
**Pull request by Cioraz**
## Type of Change

- [ ] Bugfix
- [ ] New feature
- [ ] Enhancement
- [x] Refactoring
- [ ] Dependency updates
- [ ] Documentation
- [ ] CI/CD

## Description

Handling the unhandled variants for trustpay
Issue #2287

### Additional Changes

- [ ] This PR modifies the API contract
- [ ] This PR modifies the database schema
- [ ] This PR modifies application configuration/environment variables

## Checklist

- [x] I formatted the code `cargo +nightly fmt --all`
- [ ] I addressed lints thrown by `cargo clippy`
- [x] I reviewed the submitted code
- [ ] I added unit tests for my changes where possible
- [ ] I added a [CHANGELOG](/CHANGELOG.md) entry if applicable

### CODE REVIEW

**Comment by srujanchikke** [comment]
Hey @Cioraz ,

Could you please attach the issue that you're working on ?

**Comment by Cioraz** [comment]
Hey,
Do you mean to write the issue number in the commit message?

File: crates/router/src/connector/trustpay/transformers.rs [hunk 1]
```diff
@@ -238,7 +238,20 @@ impl TryFrom<&BankRedirectData> for TrustpayPaymentMethod {
...
+            | api_models::payments::BankRedirectData::OnlineBankingFinland { .. }
+            | api_models::payments::BankRedirectData::OnlineBankingPoland { .. }
+            | api_models::payments::BankRedirectData::OnlineBankingSlovakia { .. }
+            | api_models::payments::BankRedirectData::OpenBankingUk { .. }
+            | api_models::payments::BankRedirectData::Przelewy24 { .. }
+            | api_models::payments::BankRedirectData::Trustly { .. }
+            | api_models::payments::BankRedirectData::OnlineBankingFpx { .. }
+            | api_models::payments::BankRedirectData::OnlineBankingThailand { .. } => {
+                Err(errors::ConnectorError::NotImplemented("Payment methods".to_string()).into())
+            }
```
Comment (srujanchikke) [inline crates/router/src/connector/trustpay/transformers.rs:254]: Same here use this function`get_unimplemented_payment_method_error_message ` from utils

File: crates/router/src/connector/trustpay/transformers.rs [hunk 2]
```diff
@@ -419,7 +432,19 @@ impl TryFrom<&TrustpayRouterData<&types::PaymentsAuthorizeRouterData>> for Trust
...
+            | api::PaymentMethodData::PayLater(_)
+            | api::PaymentMethodData::BankDebit(_)
+            | api::PaymentMethodData::BankTransfer(_)
+            | api::PaymentMethodData::Crypto(_)
+            | api::PaymentMethodData::MandatePayment
+            | api::PaymentMethodData::Reward
+            | api::PaymentMethodData::Upi(_)
+            | api::PaymentMethodData::Voucher(_)
+            | api::PaymentMethodData::GiftCard(_) => {
+                Err(errors::ConnectorError::NotImplemented("Payment methods".to_string()).into())
```
Comment (srujanchikke) [inline crates/router/src/connector/trustpay/transformers.rs:446]:
```suggestion
                Err(errors::ConnectorError::NotImplemented(
                        utils::get_unimplemented_payment_method_error_message("trustpay"),
                    )
                    .into())
```

Use this function from utils

**Comment by Cioraz** [comment]
Hey,
I had a small doubt here when you use utils:: <is this any message or is this a implementation in the utils package?>

**Comment by Cioraz** [comment]
Got my error will send the final PR with no errors in sometime

**Review by srujanchikke — APPROVED** [review]
Looks good to me!

**Review by SanchithHegde — APPROVED** [review]
Looks good to me!

Thanks for the PR, @Cioraz!

**Comment by swangi-kumari** [comment]
Hey @Cioraz ,
Thanks a bunch for all your contributions! We've got a little something for you to show our appreciation.  
Just take a moment to fill out this [form](https://forms.gle/yqbtjFuP6oxLaBKN8), and get ready for some awesome swag coming your way.  
Thanks!

**Pull request merged**